packages/core-components/src/layout/Sidebar/SidebarSubmenu.tsx

```diff
@@ -91,10 +97,25 @@ export const SidebarSubmenu = (props: SidebarSubmenuProps) => {
   const classes = useStyles({ left: left })();
 
   const { isHoveredOn } = useContext(SidebarItemWithSubmenuContext);
+  const [isSubmenuOpen, setIsSubmenuOpen] = useState(false);
+  const hoverTimerRef = useRef<number>();
+
+  useEffect(() => {
```

Comment:
Is there anyway we could do this in CSS instead of doing it in Javascript? I'm worried that these things might not get cleaned up properly and could lead to some pretty unintuitive error messages  from React.

Reply:
I was able to minimise the use of js for this, but useEffect is necessary as the transition wouldn't apply using just `isHoveredOn`.

Reply:
Perfect! Much better! Just really didn't want to have all of the `setTimeouts` as they can be messy to clean up :)